pkg/webhook/webhook.go

```diff
@@ -175,3 +192,104 @@ func (c *Webhook) validateHandlerFunc(review *admission_v1beta1.AdmissionReview)
 		Allowed: true,
 	}
 }
+
+func (c *Webhook) validateRelease(request *admission_v1beta1.AdmissionRequest, release shipper.Release) error {
+	var err error
+	overrideRBs := rolloutblock.NewObjectNameList(release.Annotations[shipper.RolloutBlocksOverrideAnnotation])
+	err = c.validateOverrideRolloutBlockAnnotation(overrideRBs, release.Namespace)
+	if err != nil {
+		return err
+	}
+
+	switch request.Operation {
+	case kubeclient.Create:
+		err = c.processRolloutBlocks(release.Namespace, overrideRBs)
+	case kubeclient.Update:
+		var oldRelease shipper.Release
+		err = json.Unmarshal(request.OldObject.Raw, &oldRelease)
+		if err != nil {
+			return nil
+		}
+		if !reflect.DeepEqual(release.Spec, oldRelease.Spec) {
+			err = c.processRolloutBlocks(release.Namespace, overrideRBs)
+		}
+	default:
+	}
+
+	return err
+}
+
+func (c *Webhook) validateApplication(request *admission_v1beta1.AdmissionRequest, application shipper.Application) error {
+	var err error
+	overrideRBs := rolloutblock.NewObjectNameList(application.Annotations[shipper.RolloutBlocksOverrideAnnotation])
+	err = c.validateOverrideRolloutBlockAnnotation(overrideRBs, application.Namespace)
+	if err != nil {
+		return err
+	}
+
+	switch request.Operation {
+	case kubeclient.Create:
+		err = c.processRolloutBlocks(application.Namespace, overrideRBs)
+	case kubeclient.Update:
+		var oldApp shipper.Application
+		err = json.Unmarshal(request.OldObject.Raw, &oldApp)
+		if err != nil {
+			return nil
+		}
+		if !reflect.DeepEqual(application.Spec, oldApp.Spec) {
+			err = c.processRolloutBlocks(application.Namespace, overrideRBs)
+		}
+	default:
+	}
+
+	return err
+}
+
+func (c *Webhook) processRolloutBlocks(namespace string, overrideRBs rolloutblock.ObjectNameList) error {
+	existingRBs := rolloutblock.NewObjectNameListFromRolloutBlocksList(c.existingRolloutBlocks(namespace))
+
+	nonOverriddenRBs := existingRBs.Diff(overrideRBs)
+	if len(nonOverriddenRBs) > 0 {
+		return shippererrors.NewRolloutBlockError(nonOverriddenRBs.String())
+	}
+
+	return nil
+}
+
+func (c *Webhook) existingRolloutBlocks(namespace string) []*shipper.RolloutBlock {
```

Comment:
It seems like we're not handling any of the errors in this function. That's dangerous because it might actually allow rollouts through if they're blocked, but an error happens here.